feat: code mod avm changes

Kingshuk-Microsoft opened the pull request:
## Description

This pull request updates the "Modernize Your Code" Solution Accelerator module and its associated workflow. The main changes include compliance improvements with AVM standards, removal of support for certain existing resource IDs, updates to API versions, and workflow configuration adjustments.

**Module improvements and compliance:**

* Fixed Bicep linter warnings (`no-unused-params`, `use-recent-api-versions`) and updated parameter descriptions to comply with AVM standards (Required./Optional. prefix and trailing dot).
* Removed support for existing Log Analytics and AI Project resource IDs by deleting the `azureExistingAIProjectResourceId` and `existingLogAnalyticsWorkspaceId` parameters. **This is a breaking change.**
* Updated API versions for Key Vault and Cognitive Services resources.

**Documentation updates:**

* Revised `README.md` to clarify the module’s purpose and target audience, emphasizing its status as a CSA CTO Gold Standard Solution Accelerator.
* Added version `0.1.1` and corresponding changes to `CHANGELOG.md`.

**Workflow configuration:**

* Updated workflow triggers to match the new module path and workflow file, replacing references to deprecated ML Hub dependencies.
* Changed the GitHub Actions checkout version from v5 to v4 for compatibility.

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
[![avm.ptn.sa.modernize-your-code](https://github.com/Ravikirana-Microsoft/bicep-registry-modules/actions/workflows/avm.ptn.sa.modernize-your-code.yml/badge.svg)](https://github.com/Ravikirana-Microsoft/bicep-registry-modules/actions/workflows/avm.ptn.sa.modernize-your-code.yml)


## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation
- [ ] Update to CI Environment or utilities (Non-module affecting changes)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] I have updated the module's CHANGELOG.md file with an entry for the next version

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

Ravikirana-Microsoft committed:
basic avm code for byoc,dkm and cwyd added

Ravikirana-Microsoft committed:
basic code reverted

Ravikirana-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Ravikirana-Microsoft committed:
avm coe updated to latest

Ravikirana-Microsoft committed:
iisues fixed

Ravikirana-Microsoft committed:
version updated

Ravikirana-Microsoft committed:
issues fix

Ravikirana-Microsoft committed:
pipeline issues fix

Ravikirana-Microsoft committed:
pipeline build issues fix

Ravikirana-Microsoft committed:
build error fix

Ravikirana-Microsoft committed:
changes reverted

Ravikirana-Microsoft committed:
Data Collection issue fix

Ravikirana-Microsoft committed:
build error fixes

Ravikirana-Microsoft committed:
issues fix

Ravikirana-Microsoft committed:
folder renamed

Ravikirana-Microsoft committed:
comments removed

Ravikirana-Microsoft committed:
dependency added

Ravikirana-Microsoft committed:
aindpoint changed

Ravikirana-Microsoft committed:
torage account issues fixed

Ravikirana-Microsoft committed:
pipeline issues fix

Ravikirana-Microsoft committed:
mentaince config added for vm

Ravikirana-Microsoft committed:
security issue fix

Ravikirana-Microsoft committed:
security issues fix

Ravikirana-Microsoft committed:
test case issues fix

Ravikirana-Microsoft committed:
issues fixes

Ravikirana-Microsoft committed:
deploy issue fix

AjitPadhi-Microsoft committed:
updated macae modules

AjitPadhi-Microsoft committed:
updated change log

Ravikirana-Microsoft committed:
waf-deploy issue fix

Ravikirana-Microsoft committed:
static validation fail issue fix

AjitPadhi-Microsoft committed:
removed branch changes

AjitPadhi-Microsoft committed:
Merge pull request #27 from Ravikirana-Microsoft/avm-macae

Avm macae

AjitPadhi-Microsoft committed:
fixed location issue

Ravikirana-Microsoft committed:
waf-deply failed fix

AjitPadhi-Microsoft committed:
added ckm updated modules

AjitPadhi-Microsoft committed:
removed branch from ci

Ravikirana-Microsoft committed:
build error fix

AjitPadhi-Microsoft committed:
updated test file

AjitPadhi-Microsoft committed:
validation issue fix

AjitPadhi-Microsoft committed:
fixed pipeline issue

AjitPadhi-Microsoft committed:
test file updated

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
test file updated

AjitPadhi-Microsoft committed:
test file updated

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
Avm upgraded changes for content processing

AjitPadhi-Microsoft committed:
test case updated

Abdul-Microsoft committed:
AVM-RES: Update Content Processing Solution Accelerator

Abdul-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Abdul-Microsoft committed:
fixed the pipeline issue for content processing

Abdul-Microsoft committed:
fix pipeline issue

Abdul-Microsoft committed:
fix for static validation issue

AjitPadhi-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Abdul-Microsoft committed:
deploymentfix

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
vm & cr deployment fix

Abdul-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

AjitPadhi-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
deployment fix

AjitPadhi-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
deployment fix for content processing

Abdul-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

AjitPadhi-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
AVM-RES: Update Container Registry module for private networking

Abdul-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

AjitPadhi-Microsoft committed:
deployment script added

AjitPadhi-Microsoft committed:
deployment script added

AjitPadhi-Microsoft committed:
deployment script fixed

AjitPadhi-Microsoft committed:
deployment script fixed

Abdul-Microsoft committed:
deployment fix for content processing

Abdul-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Abdul-Microsoft committed:
changed location for testing

Harsh-Microsoft committed:
refactored dkm avm sa implementation

* Refactor network modules: Remove old bastionHost, jumpbox, and network configurations; introduce a new consolidated virtualNetwork module with enhanced subnet and NSG definitions.

* CHANGE: Update version and changelog for Document Knowledge Mining

- Updated version to 0.2 in version.json
- Changed changelog version from 0.1.0 to 0.2.0

* Refactor code structure for improved readability and maintainability

* Removed 'hb-dkm-update' branch from the push triggers in the document knowledge mining workflow.

* Adjusted default values for several parameters to false for clarity

* Update resource types and references in README

* Update solutionUniqueToken length to 5

AjitPadhi-Microsoft committed:
deployment script fixed

AjitPadhi-Microsoft committed:
updated test file

Abdul-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Ravikirana-Microsoft committed:
post deployement running issue fix

Ravikirana-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Abdul-Microsoft committed:
deployment fix

Abdul-Microsoft committed:
changed the solution name for testing

Harsh-Microsoft committed:
renamed folder enterprise-grade to waf-aligned configuration.

Harsh-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Abdul-Microsoft committed:
deployment fix

Harsh-Microsoft committed:
AVM-RES: Update container registry module parameters

- Added parameters for redundancy and secondary location in the container registry module.
- Updated the SKU selection logic to support redundancy.
- Changed the module version from 0.9.1 to 0.9.3.

Harsh-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Harsh-Microsoft committed:
- Updated default parameters for enterprise-grade configuration.
- Removed redundancy parameters from container registry module.
- Added new test for enterprise-grade configuration deployment.

Ravikirana-Microsoft committed:
code reverted

Ravikirana-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Abdul-Microsoft committed:
removed the uselocalbuild flag

AjitPadhi-Microsoft committed:
deployment script fixed

AjitPadhi-Microsoft committed:
minor update test file

AjitPadhi-Microsoft committed:
updated bicep

AjitPadhi-Microsoft committed:
static issue fix

Abdul-Microsoft committed:
removed testing code

Ravikirana-Microsoft committed:
image tag chnaged

Ravikirana-Microsoft committed:
Merge pull request #29 from Ravikirana-Microsoft/byoc-upgrade

fix: image tag changed

AjitPadhi-Microsoft committed:
updated bicep

Harsh-Microsoft committed:
Remove solutionName parameter from test config

Harsh-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

AjitPadhi-Microsoft committed:
bicep issue fix

AjitPadhi-Microsoft committed:
bicep issue fix

AjitPadhi-Microsoft committed:
bicep issue fix

AjitPadhi-Microsoft committed:
bicep issue fix

AjitPadhi-Microsoft committed:
bicep issue fix

AjitPadhi-Microsoft committed:
bicep issue fix

AjitPadhi-Microsoft committed:
bicep issue fix

Abdul-Microsoft committed:
added param foe image tag

Abdul-Microsoft committed:
updated readme and main.json

Abdul-Microsoft committed:
removed exp code

AjitPadhi-Microsoft committed:
minor fix

Abdul-Microsoft committed:
deployment fix for vm

Abdul-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

AjitPadhi-Microsoft committed:
psrule issue fix

AjitPadhi-Microsoft committed:
psrule issue fix

AjitPadhi-Microsoft committed:
psrule issue fix

Ravikirana-Microsoft committed:
for skuCapacity enablesaclability condition removed

Ravikirana-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

AjitPadhi-Microsoft committed:
psrule issue fix

Ravikirana-Microsoft committed:
changes reverted due to pipeline failures

Ravikirana-Microsoft committed:
Merge branch 'main' of https://github.com/Ravikirana-Microsoft/bicep-registry-modules

Kingshuk-Microsoft committed:
avm changes

Kingshuk-Microsoft committed:
code mod avm changes

Kingshuk-Microsoft committed:
AVM-PTN: Add DeploymentName to resource tags

- Included DeploymentName in the tags for resources to enhance tracking and identification.
- Updated templateHash in metadata for consistency with the latest changes.

Prajwal-Microsoft commented on avm/ptn/sa/modernize-your-code/modules/network/network-resources.bicep:
```diff
@@ -0,0 +1,121 @@
+@minLength(6)
```
This file is not required

Kingshuk-Microsoft replied:
removed

Kingshuk-Microsoft commented on avm/ptn/sa/modernize-your-code/main.bicep:
```diff
@@ -59,54 +55,78 @@ param enableScaling bool = false
+
+@minLength(1)
+@description('Optional. Set the Image tag. Defaults to latest_2025-09-22_455.')
+param imageVersion string = 'latest_2025-09-22_455'
```
updated

Ravikirana-Microsoft committed:
latest BYOC code

Ravikirana-Microsoft committed:
build error fix

Ravikirana-Microsoft committed:
read me updated

Kingshuk-Microsoft committed:
Remove deprecated network resources and virtual network modules to streamline infrastructure code.

Kingshuk-Microsoft commented on avm/ptn/sa/modernize-your-code/modules/network/jumpbox.bicep:
```diff
@@ -18,7 +18,6 @@ import { subnetType } from 'virtualNetwork.bicep'
 @description('Optional. Subnet configuration for the Jumpbox VM.')
```
removed

Kingshuk-Microsoft committed:
AVM-PTN: Update version to 0.2

AjitPadhi-Microsoft committed:
Merge branch 'Azure:main' into main

Kingshuk-Microsoft committed:
AVM-PTN: Add branch for modernize-your-code workflow

- Added 'avm/ptn/sa/modernize-your-code' branch to the push event triggers.

Kingshuk-Microsoft committed:
AVM-PTN: Remove redundant branch from push triggers

Kingshuk-Microsoft committed:
AVM-PTN: Update changelog version to 0.2.0

Abdul-Microsoft committed:
avm upgrade changes for cwyd

Abdul-Microsoft committed:
avm upgrade changes

Abdul-Microsoft committed:
CHANGE: Update CHANGELOG and README for clarity

- Added detailed examples in README for better user guidance.
- Refined descriptions in main.bicep for parameter clarity.
- Updated CHANGELOG to reflect recent changes and improvements.

Abdul-Microsoft committed:
avm pipeline fix

Abdul-Microsoft committed:
updated version.json

Kingshuk-Microsoft committed:
AVM-PTN: Refactor location handling in Bicep modules

Abdul-Microsoft committed:
Updated description in main.json to include post-deployment instructions.

Harsh-Microsoft committed:
Update solutionName parameter to use placeholder

Abdul-Microsoft committed:
Added azureOpenAIEmbeddingModelCapacity and azureOpenAIModelCapacity parameters to main.bicep and README.md.
- Removed existingLogAnalyticsWorkspaceId parameter from main.bicep and related files.
- Updated monitoring.bicep to streamline workspace resource ID handling.
- Adjusted test cases in sandbox and WAF-aligned configurations to include new parameters.

Abdul-Microsoft committed:
Updated the image tag

NirajC-Microsoft committed:
Update main.json with bicep changes

Abdul-Microsoft committed:
Merge pull request #32 from Ravikirana-Microsoft/avmupgrade_cwyd

fix: Merging the avm module cwyd module changes

Commit:
Merge branch 'main' into avm/ptn/sa/modernize-your-code

NirajC-Microsoft committed:
Changes AZURE-AI-AGENT-ENDPOINT Logic

Commit:
refacttor: Modified the name of service short

Commit:
feat: Set model capacity

Commit:
fix: Removed the special character

Kingshuk-Microsoft committed:
Merge remote-tracking branch 'origin/main' into avm/ptn/sa/modernize-your-code
